test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -179,10 +188,107 @@ public void failoverTest() throws InterruptedException, ExecutionException {
+        // Connect with controller instance
+        final Controller controller = new ControllerImpl(
+                ControllerImplConfig.builder()
+                        .clientConfig(clientConfig)
+                        .build(), executorService);
+
+        // Create scope, stream, and a transaction with high timeout value.
+        controller.createScope(scope).join();
+        log.info("Scope {} created successfully", scope);
+
+        createStream(controller, scope, stream, ScalingPolicy.fixed(initialSegments));
+        log.info("Stream {}/{} created successfully", scope, stream);
+
+        StreamImpl stream1 = new StreamImpl(scope, stream);
+
+        // Initiate scale operation. It will block until ongoing transaction is complete.
+        controller.startScale(stream1, segmentsToSeal, newRangesToCreate);
+
+        // Initiate and Create RG instances
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
+        readerGroupManager.createReaderGroup(readerGroupName,
+                ReaderGroupConfig.builder().stream(io.pravega.client.stream.Stream.of(scope, stream)).build());
+
+        int startInclusive = 1;
+        int endExclusive = 500;
+        log.info("Write events with range [{},{})", startInclusive, endExclusive);
+        writeEvents(scope, IntStream.range(startInclusive, endExclusive).boxed().collect(Collectors.toList()), stream);
+
+        // Now stop the controller instance executing scale operation.
+        Futures.getAndHandleExceptions(controllerService.scaleService(0), ExecutionException::new);
+        log.info("Successfully stopped one instance of controller service");
+
+        // restart controller service
+        Futures.getAndHandleExceptions(controllerService.scaleService(1), ExecutionException::new);
+        log.info("Successfully stopped one instance of controller service");
+
+        List<URI> controllerUris = controllerService.getServiceDetails();
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = controllerUris.stream().filter(ISGRPC).map(URI::getAuthority)
+                .collect(Collectors.toList());
+
+        controllerURIDirect = URI.create((Utils.TLS_AND_AUTH_ENABLED ? TLS : TCP) + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+
+        ClientConfig clientConf = Utils.buildClientConfig(controllerURIDirect);
+        // Connect to another controller instance.
+        @Cleanup
+        final Controller controller2 = new ControllerImpl(
+                ControllerImplConfig.builder()
+                        .clientConfig(clientConf)
+                        .build(), executorService);
+
+        // Note: if scale does not complete within desired time, test will timeout.
+        boolean scaleStatus = controller2.checkScaleStatus(stream1, 0).join();
```

Comment:
How is this test different from the already existing test case that checks that the scale operation succeeds across a Controller restart?